src/NumericalAlgorithms/Spectral/Projection.hpp

```diff
@@ -53,6 +53,18 @@ bool needs_projection(const Mesh<Dim>& mesh1, const Mesh<Dim>& mesh2,
  * sometimes referred to as a "restriction operator" and the
  * `projection_matrix_parent_to_child` as a "prolongation operator".
  *
+ * \par Massive quantities
+ * If the quantity that should be projected is not a function over the
+ * computational grid but a "massive" residual, i.e. a quantity
+ * \f$\int_{\Omega_k} f(x) \psi_p(x) \mathrm{d}V\f$ where \f$\psi_p\f$ are the
+ * basis functions on the mesh, then pass `true` for the parameter `massive`
+ * (default is `false`). The restriction operator for this case is just the
+ * transpose of the prolongation operator, i.e. just an interpolation matrix
+ * transpose. Note that the "massive" residual already takes the difference
+ * in element size between parent and children into account by including a
+ * Jacobian in the volume element of the integral.
```

Comment:
are you certain this holds for curved meshes? I think all our projection currently isn't conservative on curved meshes (when `u J` doesn't fit in the basis to begin with), but I don't know how that translates to "massive" quantities

Reply:
Projecting a massive quantity essentially means we project `u J`, see e.g. `InitializeFixedSources.hpp` or `DgOperator.hpp` where I multiply by the Jacobian. Not sure if that's conservative, but it's better than what we currently do when projecting between faces and mortars where (I think) we do a "logical" projection, i.e. only have a factor of 2 from the Jacobian between parent and children logical coords. That's correct on a rectilinear grid, but I think for better projections on a curved mesh we should remove the factor of 2 and multiply by Jacobians. Of course that means we have to evaluate and store the Jacobians on the mortars.

Reply:
ah okay, yea if you project `u J` then it's conservative, or, you conserve the inertial frame quantity, which is generally what we want :)